src/GraphQL/Mutation/MutationType.php

```diff
@@ -878,18 +881,34 @@ public function buildCreateAssetMutation(&$config, $context)
                     $newInstance = new $className();
                     $newInstance->setParentId($parent->getId());
 
+                    $tags = [];
                     if (isset($args['input'])) {
                         $inputValues = $args['input'];
                         foreach ($inputValues as $key => $value) {
-                            if ($key === 'data') {
-                                $value = base64_decode($value);
+                            //TODO: ask pimcore/pimcore to implement something like Asset::setTags
+                            if ($key == 'tags') {
+                                $tags = $me->getTagsFromInput($value);
+                                if (false === $tags) {
+                                    return [
+                                        'success' => false,
+                                        'message' => 'no "id" nor "path" tag data defined for tag, or tag not found',
+                                    ];
+                                }
+                            } else {
+                                if ($key === 'data') {
+                                    $value = base64_decode($value);
+                                }
+                                $setter = 'set' . ucfirst($key);
+                                $newInstance->$setter($value);
```

Comment:
BTW, to make the code more clean it would be great to have something like `Asset::setTags` implemented in pimcore/pimcore. If so, tags creation and update logic can be fully moved to Asset out of data-hub.

Reply:
yes, but please consider the minimum requirements:

  "require": {
    "pimcore/pimcore": "^6.9 || ^10.0",